### ISSUE TITLE
docs: documentation clarity and accuracy improvements across Home, Getting Started, and Tutorial

### ISSUE THREAD

**Issue by sriumcp** [inference-sim/inference-sim#462]
## Summary

Documentation audit identified 17 issues across Home, Getting Started, User Guide, Contributing, and site navigation. Themes: overly specific language that will become stale, underselling key features, misleading descriptions, DRY violations, missing pages, and navigation that doesn't match the learning path.

---

### 1. Overly specific latency model description

Line 3 says _"all driven by trained performance coefficients or analytical roofline estimates"_. Too specific — will become stale as new latency backends are added. The `LatencyModel` interface + `NewLatencyModelFunc` registration pattern already supports arbitrary backends.

**Fix:** Replace with _"pluggable latency models (data-driven coefficients, analytical roofline, or custom backends)"_.

### 2. Workload spec undersold as "ServeGen-informed"

Line 30 says _"ServeGen-informed workload generation with multi-client traffic classes"_, which buries one of BLIS's strongest features. The canonical workload spec is a first-class YAML DSL supporting:

- Multiple client populations with independent arrival processes (Poisson, Gamma, Weibull, constant)
- 5 distribution types (Gaussian, exponential, Pareto-lognormal, constant, empirical)
- SLO classes (critical, standard, sheddable, batch, background), prefix groups, cohort dynamics (diurnal/spike/drain), multimodal, reasoning multi-turn
- Composable specs via `blis compose`
- Built-in presets (bursty, unfair, prefix-heavy, mixed-slo)

**Fix:** Give the workload spec its own prominent bullet rather than framing it as a ServeGen footnote.

### 3. Metrics collection not highlighted as a feature

BLIS collects a rich three-level metrics pipeline that deserves a feature bullet:

- **Per-request**: TTFT, ITL, E2E, scheduling delay, SLO class, tenant ID, assigned instance, model tag, prefill/decode token counts
- **Per-instance aggregate**: TTFT/ITL/E2E at mean/p90/p95/p99, throughput (req/s and tokens/s), KV cache (preemptions, cache hit rate, thrashing rate, allocation failures), queue depth time series, batch size time series, dropped unservable
- **Cluster-level**: all of the above aggregated + anomaly detection (priority inversions, HOL blocking events), per-SLO-class distributions, Jain fairness index, SLO attainment scores, multi-objective fitness evaluation

**Fix:** Add a feature bullet. Rename existing `docs/guide/results.md` to "Metrics & Results" for discoverability.

---

### 4. "Not a load generator" is inaccurate

`cmd/observe.go` has a `RealClient` that sends real HTTP requests to OpenAI-compatible endpoints (streaming + non-streaming) with trace v2 recording. Exists but not fully integrated.

**Fix:** Soften to: _"Not primarily a load generator — BLIS focuses on simulation. Real-mode traffic generation against OpenAI-compatible endpoints is experimental. For production load testing, use inference-perf or genai-perf."_

### 5. "Not cycle-accurate" — remove

Positions a current implementation detail as a fundamental limitation. Implied by "simulator."

**Fix:** Remove entirely.

### 6. "Homogeneous instances only" — remove

Mixed-fleet modeling is near-future work. The `WorkloadSpec` already has a per-client `model` field.

**Fix:** Remove entirely.

### 7. Key Capabilities overlap with Home features (DRY violation)

Both `docs/index.md` (Key Features) and `docs/getting-started/index.md` (Key Capabilities) maintain separate feature lists that can drift.

**Fix:** Home page owns the authoritative feature list. Getting Started links to it and focuses on positioning (Who/Why/What-Not).

---

### 8. Remove standalone prefix-affinity routing policy

The standalone `prefix-affinity` routing policy (`routing.go:206-248`) should be removed entirely — code, factory, validation map, docs, and tests.

- **Superseded by the scorer** in `WeightedScoring` — proportional match ratio, naturally composed with load-balancing
- **Known degenerate behavior (H21)**: cold-start ties pile ALL requests onto instance 0
- **Binary sticky mapping is fundamentally limited**: unbounded `prefixMap`, no LRU, no partial matches

**Files:** `sim/routing.go`, `sim/bundle.go`, `docs/guide/routing.md`, related tests

### 9. Signal Freshness section is misleading about "Immediate" mode

"Immediate" implies the router has instant access to instance-level state. In the DES, it's just a struct field read.

**Fix:** Rewrite to lead with modeling rationale, clarify DES semantics, separate user concept from implementation. Note #463 tracks making QueueDepth/BatchSize realistically stale.

---

### 10. Extract Scheduling & Priority (`docs/guide/scheduling.md`)

Buried under Routing (`routing.md:67-78`). Separate architectural concern.

**Content:** 4 schedulers, 3 priority policies, their interaction, SLO-class patterns, "when to use which."

### 11. Extract Admission Control (`docs/guide/admission.md`)

Buried in Cluster page (`cluster.md:66-86`). Same pluggable policy pattern as routing/scheduling.

**Content:** 3 strategies, token bucket mechanics, admission/routing latency modeling, when to use it.

### 12. Rename "Roofline Mode" to "Latency Models" (`docs/guide/latency-models.md`)

Current page is roofline-only, treats blackbox as footnote. Users ask "how does BLIS estimate latency?"

**Content:** What are latency models, blackbox mode (default), roofline mode, when to use which, pluggable architecture.

### 17. Add "Skills & Plugins" page (`docs/guide/skills-and-plugins.md`)

BLIS uses skills and plugins from four sources for both end-user workflows (experimentation, research) and contributor workflows (PR review, planning). No single page documents the ecosystem.

| Layer | Location | Audience | Examples |
|-------|----------|----------|----------|
| **Project skills** | `.claude/skills/` | All users | `convergence-review`, `hypothesis-experiment` |
| **BLIS SDLC plugins** | [`sdlc-plugins`](https://github.com/inference-sim/sdlc-plugins) | All users | `research-ideas`, `hypothesis-test` |
| **General superpowers** | [`superpowers-marketplace`](https://github.com/obra/superpowers-marketplace) | All users | `superpowers` (TDD, debugging, plans, worktrees), `episodic-memory` |
| **Official plugins** | [`claude-plugins-official`](https://github.com/anthropics/claude-plugins-official) | All users | `commit-commands`, `pr-review-toolkit`, `code-review`, `feature-dev` |
| **Community plugins** | [`awesome-claude-plugins`](https://github.com/ComposioHQ/awesome-claude-plugins) | All users | `audit-project`, `bug-fix`, `debugger` |

**Content:** What's available, marketplace links, installation, workflow mapping, project-level vs user-level skills.

**Nav:** Under User Guide, after Experimentation.

---

### 13. Rename "Interpreting Results" to "Metrics & Results"

Current title not discoverable. Content is comprehensive but title undersells it.

### 14. Workloads page needs user-story framing and missing features

Reads bottom-up (syntax reference) instead of top-down (how do I model chatbot traffic?).

**Fix:** Add "Modeling Real Workloads" section as centerpiece:
- Interactive chat → Poisson arrivals, Gaussian tokens, `critical` SLO, `prefix_group`
- RAG with shared prefixes → `prefix_group` + `prefix_length`, prefix-affinity scorer
- Batch/offline → `batch` SLO, high token counts, `gamma` arrivals
- Bursty flash-sale → `gamma` with high CV, or cohort `spike` pattern
- Multi-turn conversations → `reasoning` field with context accumulation

Plus missing coverage: cohorts, `blis compose`, multimodal, reasoning, presets.

---

### 15. Step 1 is tightly coupled to blackbox latency model

Hand-calculates from beta coefficients — wrong for roofline users and future backends.

**Fix:** Reframe as "Estimate Instance Capacity" — run at low load empirically, link to Latency Models page.

---

### 16. Move Concepts before User Guide

**Fix:** Home → Getting Started → **Concepts** → **User Guide** → Reference → Methodology → Contributing

**Target nav:**

```yaml
nav:
  - Home: index.md
  - Getting Started:
      - getting-started/index.md
      - Installation: getting-started/installation.md
      - Quick Start: getting-started/quickstart.md
      - "Tutorial: Capacity Planning": getting-started/tutorial.md
  - Concepts:
      - concepts/index.md
      - Glossary: concepts/glossary.md
      - Cluster Architecture: concepts/architecture.md
      - Core Engine: concepts/core-engine.md
      - Roofline Estimation: concepts/roofline.md
  - User Guide:
      - guide/index.md
      - Routing Policies: guide/routing.md
      - Admission Control: guide/admission.md           # NEW
      - Scheduling & Priority: guide/scheduling.md      # NEW
      - Latency Models: guide/latency-models.md          # RENAMED from roofline.md
      - KV Cache & Memory: guide/kv-cache.md
      - Workload Specifications: guide/workloads.md
      - Cluster Simulation: guide/cluster.md
      - Metrics & Results: guide/results.md              # RENAMED
      - Hypothesis Experimentation: guide/experimentation.md
      - Skills & Plugins: guide/skills-and-plugins.md    # NEW
  - Reference:
      - reference/index.md
      - Configuration: reference/configuration.md
      - Supported Models: reference/models.md
      - Workload Spec Schema: reference/workload-spec.md
  - Methodology:
      - methodology/index.md
      - Strategy Evolution: methodology/strategy-evolution.md
      - Discovered Principles: methodology/principles.md
  - Contributing:
      - contributing/index.md
      - Extension Recipes: contributing/extension-recipes.md
      - PR Workflow: contributing/pr-workflow.md
      - Design Process: contributing/design-process.md
      - Macro Planning: contributing/macro-planning.md
      - Hypothesis Experiments: contributing/hypothesis.md
      - Convergence Protocol: contributing/convergence.md
      - Standards: ...
      - Templates: ...
```

---

## Files to modify

| File | Issues |
|------|--------|
| `docs/index.md` | 1, 2, 3 |
| `docs/getting-started/index.md` | 4, 5, 6, 7 |
| `docs/getting-started/tutorial.md` | 15 |
| `docs/guide/routing.md` | 8, 9, 10 (remove scheduling section) |
| `docs/guide/scheduling.md` (new) | 10 |
| `docs/guide/admission.md` (new) | 11 |
| `docs/guide/latency-models.md` (new, replaces roofline.md) | 12 |
| `docs/guide/cluster.md` | 11 (remove admission section) |
| `docs/guide/results.md` | 13 (rename in nav) |
| `docs/guide/workloads.md` | 14 |
| `docs/guide/skills-and-plugins.md` (new) | 17 |
| `mkdocs.yml` | 10, 11, 12, 13, 16, 17 |
| `sim/routing.go` | 8 (remove PrefixAffinity policy) |
| `sim/bundle.go` | 8 (remove from validRoutingPolicies) |

## Related issues

- #463 — Model QueueDepth and BatchSize with realistic staleness (referenced by issue 9)

**Comment by sriumcp** [comment]
## Additional item

**14. Rename "Roofline Mode" to "Latency Models" and cover both modes as peers**

The current `docs/guide/roofline.md` page is roofline-only and treats blackbox as a footnote in a comparison table. From a user's perspective, the first question is "how does BLIS estimate latency?" — both modes are answers.

**Fix:** Rename to "Latency Models" (`docs/guide/latency-models.md`) and restructure:

1. **What are latency models?** — the `LatencyModel` interface determines step time; BLIS ships two backends
2. **Blackbox mode** (default) — trained alpha/beta coefficients from `defaults.yaml`, higher tail accuracy, which models have pre-trained coefficients
3. **Roofline mode** — analytical FLOPs/bandwidth, `--roofline` flag, auto-fetch, manual config, adding new models
4. **When to use which** — existing comparison table, expanded
5. **Pluggable architecture** — brief note that the interface supports custom backends (forward-looking, aligns with issue 1's "pluggable latency models" framing)

This also makes the Tutorial fix (issue 11) cleaner — Step 1 can say "run at low load to measure throughput" and link to the Latency Models page for understanding why.

**Files:** rename `docs/guide/roofline.md` → `docs/guide/latency-models.md`, update `mkdocs.yml` nav entry, update cross-references.
